lib/mongo/client.rb

```diff
@@ -855,14 +857,32 @@ def validate_new_options!(opts = Options::Redacted.new)
     end
 
     # Validates all options after they are set on the client.
-    # This method is intended to catch combinations of options which do are
+    # This method is intended to catch combinations of options which are
     # not allowed.
     def validate_options!
       if options[:write] && options[:write_concern] && options[:write] != options[:write_concern]
         raise ArgumentError, "If :write and :write_concern are both given, they must be identical: #{options.inspect}"
       end
     end
 
+    # Validates all authentication-related options after they are set on the client
+    # This method is intended to catch combinations of options which are not allowed
+    def validate_authentication_options!
```

Comment:
It might also be good to extract this method and validate auth options passed to the `Mongo::Auth::User` object, but I think that's slightly outside the scope of [[RUBY-1745]](https://jira.mongodb.org/browse/RUBY-1745) (but open to other opinions there).